### PR TITLE
Fixes 2613: React 18/Patternfly 5/Node 18 upgrade

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

This updates our application to use React 18, Patternfly 5, and node 18. 

## Testing steps

- Ensure if checking locally you are using nvm and doing "nvm use" installing the correct new node version (if necessary)
- npm install
- Run the app as you regularly would. 
- Regression testing is basically what we're looking for.

Additional information: 

- There is a [known bug ](https://issues.redhat.com/browse/HMS-2588) with the popular repository page that stems from the API not returning an associated UUID.  This is confirmed as not UI related.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-2613

**Review by rverdile — COMMENTED** [review]
all working the same as far as I can tell

**Review by jlsherrill — APPROVED** [review]
ack from me.  i can't say i studied all lines of code change in depth, but overall it looks okay and all behavior runs as expected.

**Comment by Andrewgdewar** [comment]
Thanks for reviewing @rverdile @jlsherrill 🍻

File: src/Pages/ContentListTable/components/AddContent/AddContent.tsx [hunk 1]
```diff
@@ -535,23 +544,18 @@ const AddContent = () => {
...
-                            variant='warning'
-                            style={{ paddingTop: '10px' }}
-                            title='Disabling snapshots may result in a higher risk of losing content or unintentionally modifying it irreversibly.'
-                            isInline
-                            isPlain
-                          />
+                          <FormAlert style={{ paddingTop: '20px' }}>
+                            <Alert
+                              variant='warning'
+                              title='Disabling snapshots may result in a higher risk of losing content or unintentionally modifying it irreversibly.'
```
Comment (swadeley) [inline src/Pages/ContentListTable/components/AddContent/AddContent.tsx:550]:
```suggestion
                              title='Disabling snapshots might result in a higher risk of losing content or unintentionally modifying it irreversibly.'
```

File: src/components/Error/ErrorPage.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ interface Props {
 
 export const ErrorPage = ({ children }: Props) => (
   <ErrorBoundaryPage
-    headerTitle='Content Sources'
+    headerTitle=''
```
Comment (swadeley) [inline src/components/Error/ErrorPage.tsx:16]: it that title supposed to be blank?

File: src/components/NoPermissionsPage/NoPermissionsPage.tsx [hunk 3]
```diff
@@ -24,10 +24,11 @@ export const NoPermissionsPage: React.FunctionComponent = () => (
     </PageHeader>
     <Grid style={{ margin: '24px' }}>
       <EmptyState variant={EmptyStateVariant.full}>
-        <EmptyStateIcon icon={LockIcon} />
-        <Title headingLevel='h5' size='lg'>
-          You do not have access to Repositories
-        </Title>
+        <EmptyStateHeader
+          titleText='You do not have access to Repositories'
```
Comment (swadeley) [inline src/components/NoPermissionsPage/NoPermissionsPage.tsx:28]: I don't think use of capital R is required here.

File: src/components/Error/ErrorPage.tsx (hunk 2 above)
Comment (Andrewgdewar) [reply src/components/Error/ErrorPage.tsx:16]: yes

**Comment by swadeley** [comment]
Hi
Re the header section:
`<h1 data-ouia-component-type="PF5/Title" data-ouia-safe="true" data-ouia-component-id="OUIA-Generated-Title-1" widget-type="InsightsPageHeaderTitle" class="pf-v5-c-title pf-m-2xl">Custom Repositories</h1>`

If you could add an data-ouia-component-id that would be nice. There is no convention on the style, I think this will do:  data-ouia-component-id='custom_repositories_page-header'

**Comment by Andrewgdewar** [comment]
Let's test changes on the React18Preview branch. This branch is now out of date, I had intended to close this earlier, my apologies.

**Pull request closed without merging**